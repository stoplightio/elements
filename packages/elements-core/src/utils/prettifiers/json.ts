import { Prettifier } from './types';

export const prettifyJSON: Prettifier<unknown> = json => {
  try {
    let parsedJson = json;
    if (typeof json === 'string') {
      parsedJson = JSON.parse(json);
    }

    return JSON.stringify(parsedJson, null, 2);
  } catch {
    return String(json);
  }
};
