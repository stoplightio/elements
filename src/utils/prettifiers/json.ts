import { Prettifier } from './types';

export const prettifyJSON: Prettifier<unknown> = json => {
  try {
    if (typeof json === 'string') {
      json = JSON.parse(json);
    }

    return JSON.stringify(json, null, 2);
  } catch {
    return String(json);
  }
};
