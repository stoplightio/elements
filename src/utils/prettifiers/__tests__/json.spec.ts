import { prettifyJSON } from '../json';

describe('JSON Prettifier', () => {
  test('should prettify content', () => {
    expect(prettifyJSON(`{"foo":true,"bar":false}`)).toMatchSnapshot();
  });
});
