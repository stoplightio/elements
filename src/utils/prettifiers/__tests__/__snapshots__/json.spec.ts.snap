// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON Prettifier should prettify content 1`] = `
"{
  \\"foo\\": true,
  \\"bar\\": false
}"
`;
